// auth/index.js
const AmazonCognitoIdentity = require('amazon-cognito-identity-js');

exports.handler = async function auth(event) {
    try{
        console.log('Event', event)
        const body = JSON.parse(event)
        // username & password
        var authenticationData = {
            Username : body.username,
            Password : body.password,
        };
        var authenticationDetails = new AmazonCognitoIdentity.AuthenticationDetails(authenticationData);
        // Cognito User Pool ID and App Cliend ID
        var poolData = { UserPoolId : body.userpoolid,
            ClientId : body.clientid
        };
        var userPool = new AmazonCognitoIdentity.CognitoUserPool(poolData);
        var userData = {
            Username : body.username,
            Pool : userPool
        };
        var cognitoUser = new AmazonCognitoIdentity.CognitoUser(userData);
    } catch(err){
        console.log(err);
        return err;
    }
    //return User ID Token
    return new Promise( (success, error) => {
        cognitoUser.authenticateUser(authenticationDetails, {
        onSuccess: (result) => {
            success(result.idToken.jwtToken);
        },
        onFailure: (err) => {
            console.error(err);
            error(err);
        },
    })
    });

}
